### PR TITLE
Remove ML and house numbers

### PR DESCRIPTION
**Pull request by andrew-hu**

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -15,45 +16,170 @@ app.use(express.static(__dirname + "/public" ));
...
+            console.log(err)
+        });
 
 io.on('connection',function(socket){ 
+
+    //////////////////////////////////////////////////////////////////
+    // Logging server stuff //
+    var server = net.createServer(function(sock) {
+        sock.on('err', function(err){
+            console.log("vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv\n\n");
```
Comment (jpan127) [inline server/app.js]: Keep?

File: server/app.js [hunk 2]
```diff
@@ -15,45 +16,170 @@ app.use(express.static(__dirname + "/public" ));
...
+            console.log(err);
+            console.log("\n\n^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^");
+            server.close();
+            server.listen(5002, '192.168.43.12');
```
Comment (jpan127) [inline server/app.js]: These hardcoded values should be at the top of the script or somewhere for easier configuration.

File: server/app.js [hunk 3]
```diff
@@ -15,45 +16,170 @@ app.use(express.static(__dirname + "/public" ));
...
+    var server = net.createServer(function(sock) {
+        sock.on('err', function(err){
+            console.log("vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv\n\n");
+            console.log(err);
+            console.log("\n\n^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^");
+            server.close();
+            server.listen(5002, '192.168.43.12');  
+        });
+        sock.on('data', function(data) {
+            //console.log(data);
```
Comment (jpan127) [inline server/app.js]: Instead of commenting out the logging information, would it be better to have a function:
```
function debug_log(string) {
    if (debug_on) {
        console.log(string);
     }
}
```
And have the flag `debug_on` configured at the top of the file.
This way we can turn off logging extra debug messages or not all at once.
If javascript has a simple logging module that can do this that works too.

File: server/app.js [hunk 4]
```diff
@@ -15,45 +16,170 @@ app.use(express.static(__dirname + "/public" ));
...
+            //console.log("Packet Type: ", packetType);
+            //Payload Length
+            var payloadLength = data.slice(1,2).toString();
+            payloadLength = "0x" + ascii_to_hexa(payloadLength);
+            payloadLength = parseInt(payloadLength);
+            //console.log("Payload Len: ", payloadLength);
+            //Payload
+            var payload = data.slice(2,payload).toString();
+            //console.log(payload);
+            var payload = payload.split(":");
```
Comment (jpan127) [inline server/app.js]: Doesn't this redeclare `payload`?

File: server/app.js [hunk 5]
```diff
@@ -15,45 +16,170 @@ app.use(express.static(__dirname + "/public" ));
...
+
+            //console.log("Payload: ", payload[0]);
+            //console.log("Payload: ", payload[1]);
+            //console.log("Payload: ", payload[2]);
+            console.log(packetType , payloadLength, payload);            
+            
+            if(packetType == 0 )
```
Comment (jpan127) [inline server/app.js]: Should not be `if, if, if, if`, if anything it should be `if, else if`.
Try a switch instead:
```
switch (packetType)
{
    case 0: socket.broadcast.emit; break;
    case 1: socket.broadcast.emit; break;
    case 2: socket.broadcast.emit; break;
    case 3: socket.broadcast.emit; break;
    default: break;
}
```

File: server/app.js [hunk 6]
```diff
@@ -15,45 +16,170 @@ app.use(express.static(__dirname + "/public" ));
...
+        
+        if(data.message == 20) {
+            clearInterval(repeat);
+            data.message = 14;
+        }
+
+        client.connect(5000, "192.168.43.250", function() { //192.168.43.250 port 5000
+            console.log('Connected to 192.168.43.250'); 
+            console.log('Local client sending:', data.message);
+            buf = new Buffer([data.message, 0xA, 0xFF]); // 0xA is temp value
```
Comment (jpan127) [inline server/app.js]: `0xA` and `0xFF` should be declared as constant values.
For example:
`const var temp = 0xA`
`const var dummy = 0xFF`

File: server/app.js [hunk 7]
```diff
@@ -15,45 +16,170 @@ app.use(express.static(__dirname + "/public" ));
...
+                // handle the error safely
+                console.log(err)
+            });
+        });
+    });    
+    socket.on('dutyEvent',function(data){
+        client.connect(5000, "192.168.43.250", function() { //192.168.43.250 port 5000
+            console.log('Connected to 192.168.43.250'); 
+            console.log('Local client sending duty cycle:', data.message);
+            buf = new Buffer([12, data.message, 0xFF]);
```
Comment (jpan127) [inline server/app.js]: Should use constant enumerations for the packet types.

File: server/app.js [hunk 8]
```diff
@@ -63,4 +189,219 @@ io.on('err', function(err){
 
 http.listen(port,function(){
     log.info('Server listening through port %s', port);
-});
\ No newline at end of file
+});
+
+http.on('err', function(err){
```
Comment (jpan127) [inline server/app.js]: This and below can probably be separated to a different file.

File: server/app.js [hunk 9]
```diff
@@ -63,4 +189,219 @@ io.on('err', function(err){
...
+   // handle the error safely
+    console.log(err)
+});
+
+process.on('uncaughtException', function(err) {
+  console.log('Caught exception: ' + err);
+});
+
+
+var state = ["NAVIGATING_SIDEWALK", "LOOKING_FOR_PATH" ,"FOUND_PATH_TO_HOUSE", "NAVIGATING_PATH" ,"REACHED_DOOR" , "DELIVER_PACKAGE" ]
```
Comment (jpan127) [inline server/app.js]: Const?

File: server/app.js [hunk 10]
```diff
@@ -63,4 +189,219 @@ io.on('err', function(err){
...
+    current_state = state[0];
+
+    console.log("navigating sidewalk");
+
+    var file = 'front.json';
+    jsonfile.readFile(file, function(err, obj) {
+        //console.log(obj.angle);
+
+        if(obj.angle >= 85 && obj.angle <= 95 ){
+                client.connect(5000, "192.168.43.250", function() { //192.168.43.250 port 5000
```
Comment (jpan127) [inline server/app.js]: These blocks can be a function for less repetition.

File: server/app.js [hunk 11]
```diff
@@ -63,4 +189,219 @@ io.on('err', function(err){
...
+                    });
+                });  
+
+
+    console.log("navigating_path");
+    // continue until sensors reach door
+}
+
+function reached_door() {
+    current_state = state[5];
```
Comment (jpan127) [inline server/app.js]: Instead of `state[5]` which is non-intuitive, there should be a way of explicitly selecting the state.
In C what I would do is `state[DELIVER_PACKAGE]` to specify that i want the index in the array of state that corresponds to 5=DELIVER_PACKAGE.
In Python what I would do is `state["DELIVER_PACKAGE"]` to search in the dictionary for the state.

**Review by jpan127 — CHANGES_REQUESTED** [review]

File: server/app.js (hunk 6 above)
Comment (andrew-hu) [reply server/app.js]: 0xA isn't the default for all the buffer values there , but 0xFF can be labeled as a const

File: server/app.js (hunk 9 above)
Comment (andrew-hu) [reply server/app.js]: Yeah, this will be changed to be a dict as well so constant enums can be used

File: server/app.js (hunk 10 above)
Comment (andrew-hu) [reply server/app.js]: Kinda embarassing I didn't do this earlier `:^) thanks

File: server/app.js (hunk 8 above)
Comment (andrew-hu) [reply server/app.js]: Not sure on this one, did you want the separate file to be somehow included? Might be better to keep it as is for code solidarity

**Comment by andrew-hu** [comment]
Just fixed most of the requested changes, as well as resolved merge conflicts with the master branch. Please review!

File: server/app.js (hunk 8 above)
Comment (jpan127) [reply server/app.js]: Yea, so not sure if it is typical for `app.js` to be extremely long or not.
Usually when a block of code can be contained by itself for modularity, it can be moved to a different file and included.

File: server/app.js [hunk 12]
```diff
@@ -240,158 +151,73 @@ function stop_state_machine(){
...
-                    client.write(buf);
-
-                    client.destroy();
-                    client = new net.Socket();
-                    client.on('err', function(err){
-                        // handle the error safely
-                        console.log(err)
-                    });
-                });        
+            clientConnectSendState(0,0x1E,0);
```
Comment (jpan127) [inline server/app.js]: Can these variables be named?
- What does 0 mean?
- What does 0x1E mean?
- What does 0 mean?

**Review by jpan127 — CHANGES_REQUESTED** [review]

File: server/app.js (hunk 12 above)
Comment (jpan127) [reply server/app.js]: Ditto for the other related function calls.

File: server/app.js [hunk 13]
```diff
@@ -1,246 +1,275 @@
...
+    current_state = state["NAVIGATING_SIDEWALK"];
+
+    debug_log("navigating sidewalk");
+
+    var file = 'front.json';
+    jsonfile.readFile(file, function(err, obj) {
+        //console.log(obj.angle);
+
+        if(obj.angle >= 85 && obj.angle <= 95 ){
+            clientConnectSendState(0,0x1E,0);
```
Comment (jpan127) [inline server/app.js:169]: I think we can make this stuff cleaner.
Documented in #20 .
But to merge this asap I would say do it in a separate PR.

**Review by jpan127 — APPROVED** [review]
